v3/src/transports/http/handler.ts

```diff
@@ -0,0 +1,145 @@
+import { IncomingMessage, RequestListener, ServerResponse } from "http";
+import { IHttpRequest } from "./transport";
+import { processHttpRequest } from "./transport";
+import { GraphQLSchema } from "graphql";
+import { GraphQLRequest } from "../../types";
+
+/**
+ * A factory function that receives an instance of `ApolloServer` and returns
+ * a `RequestHandler` that can be used with Node.js' `http.createServer`, or
+ * Express' `app.use`.
+ *
+ * @param apollo An instance of `ApolloServer`
+ */
+export const httpHandler: (
+  schema: GraphQLSchema,
+) => RequestListener = schema => {
+  if (!(schema instanceof GraphQLSchema)) {
+    throw new Error("Must pass a schema.");
+  }
+
+  /**
+   * Returns the handler that can be passed to the HTTP framework that
+   * respects the `(req, res)` pattern (e.g. Express or Node.js).
+   */
+  return async function httpRequestListener(req, res): Promise<void> {
+    if (!req) {
+      return responseAsInternalServerError(
```

Comment:
This feels a bit nitpicky, but this reads a bit awkwardly for me. `httpRequestListener` returns `void`, but all of these return statements _appear_ to be returning some kind of `responseXyz` (even though the functions themselves correctly return `void`). The following is just a bit more clear to me that the function is _doing something_ to response and returning nothing useful:

```typescript
responseAsInternalServerError(
  res,
  "Missing request on HTTP request handler invocation.",
);
return;
```